unittests/forked_tests_savanna.cpp

```diff
@@ -115,8 +115,10 @@ BOOST_FIXTURE_TEST_CASE(fork_with_bad_block_savanna, savanna_cluster::cluster_t)
          if (j <= i) {
             auto copy_b = std::make_shared<signed_block>(b->clone());
             if (j == i) {
-               // corrupt this block (forks[j].blocks[j] is corrupted)
-               copy_b->action_mroot._hash[0] ^= 0x1ULL;
+               // Corrupt this block (forks[j].blocks[j] is corrupted).
+               // Do not corrupt the block by modifying action_mroot, as action_mroot is checked
+               // by block header validation, _nodes[0].push_block(b) would fail.
+               copy_b->confirmed++;
```

Comment:
@greg7mdp Could you double check if this change is correct or not?

Reply:
Seems like any corrupt will work, `confirmed` is as good as anything else.

Reply:
Now it occurs to me, should we check `confirmed` is 0 for Savanna blocks?

Reply:
Yes. It is verified as part of block id, but wouldn't hurt to check it early.

Reply:
> @greg7mdp Could you double check if this change is correct or not?

Yes, seems fine to me as well.

Reply:
Thanks.

Reply:
I ended up not adding the check of `confirmed == 0`. With the check turned on, it is hard to corrupt a block in `fork_with_bad_block_savanna`. As `confirmed` is not used in Savanna and checked later on by block ID, it is OK not being checked in the header validation.